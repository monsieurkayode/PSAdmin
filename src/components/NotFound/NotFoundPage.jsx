import React from 'react';
import { string } from 'prop-types';

/**
 * @function NotFoundPage
 *
 * @param {object} props
 *
 * @returns {JSX}
 */
const NotFoundPage = ({ content }) => (
  <div style={{ marginTop: '20%' }} className="no-content">
    <h3>404! <span>| {content} not found!</span></h3>
  </div>
);

NotFoundPage.defaultProps = {
  content: 'Page'
};

NotFoundPage.propTypes = {
  content: string
};

export default NotFoundPage;
