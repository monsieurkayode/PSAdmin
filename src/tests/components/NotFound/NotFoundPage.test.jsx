import React from 'react';
import { mount } from 'enzyme';
import NotFoundPage from '../../../components/NotFound/NotFoundPage';

describe('NotFoundPage Component', () => {
  const wrapper = mount(<NotFoundPage />);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has a default content props of "Page"', () => {
    expect(wrapper.props()).toHaveProperty('content', 'Page');
  });

  it('has a single wrapping div as it roots', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });
});
