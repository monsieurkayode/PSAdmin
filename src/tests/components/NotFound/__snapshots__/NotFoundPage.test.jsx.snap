// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage Component renders correctly 1`] = `
<NotFoundPage
  content="Page"
>
  <div
    className="no-content"
    style={
      Object {
        "marginTop": "20%",
      }
    }
  >
    <h3>
      404! 
      <span>
        | 
        Page
         not found!
      </span>
    </h3>
  </div>
</NotFoundPage>
`;
